// react-svg-donut/build/ArcPath.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ArcPath;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _path = require("./path");

var _path2 = _interopRequireDefault(_path);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ArcPath(props) {
  var size = props.size,
      angle = props.angle,
      total = props.total,
      fill = props.fill,
      item = props.item,
      index = props.index,
      className = props.className,
      innerRadius = props.innerRadius,
      outerRadius = props.outerRadius,
      _onMouseEnter = props.onMouseEnter;
  var value = item.value;

  var activeAngle = Number.isNaN(value / total) || total / value === 1 ? 359.99 : value / total * 360;
  var d = (0, _path2.default)(activeAngle, angle, size, innerRadius, outerRadius);
  return _react2.default.createElement("path", {
    onMouseEnter: function onMouseEnter() {
      return _onMouseEnter(index);
    },
    onMouseLeave: function onMouseLeave() {
      return _onMouseEnter(-1);
    },
    className: className,
    d: d,
    fill: fill
  });
}

ArcPath.propTypes = {
  size: _propTypes2.default.number.isRequired,
  angle: _propTypes2.default.number.isRequired,
  total: _propTypes2.default.number.isRequired,
  fill: _propTypes2.default.string,
  item: _propTypes2.default.shape({ name: _propTypes2.default.string, value: _propTypes2.default.number }).isRequired,
  index: _propTypes2.default.number,
  className: _propTypes2.default.string,
  onMouseEnter: _propTypes2.default.func,
  innerRadius: _propTypes2.default.number.isRequired,
  outerRadius: _propTypes2.default.number.isRequired
};

ArcPath.defaultProps = {
  fill: "#ff0000",
  index: 0,
  className: "donut-chart-arcpath-0",
  // eslint-disable-next-line
  onMouseEnter: function onMouseEnter(index) {
    return console.log("Hover", index);
  }
};
