// react-svg-donut/build/Donut.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Donut;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Arcs = require("./Arcs");

var _Arcs2 = _interopRequireDefault(_Arcs);

var _InnerText = require("./InnerText");

var _InnerText2 = _interopRequireDefault(_InnerText);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Donut(props) {
  var title = props.title,
      size = props.size,
      className = props.className,
      data = props.data,
      colors = props.colors,
      innerRadius = props.innerRadius,
      outerRadius = props.outerRadius,
      activeOffset = props.activeOffset,
      onHover = props.onHover,
      active = props.active;

  var newData = data && data !== "undefined" && data.length > 0 ? data : [{ name: "", value: 1 }];
  return _react2.default.createElement(
    "svg",
    {
      className: className,
      style: { width: size + "px", height: size + "px" },
      viewBox: "0 0 " + size + " " + size
    },
    _react2.default.createElement(_Arcs2.default, {
      size: size,
      data: newData,
      active: active,
      onHover: onHover,
      className: className + "-arcs",
      colors: colors,
      innerRadius: innerRadius,
      outerRadius: outerRadius,
      activeOffset: activeOffset
    }),
    _react2.default.createElement(_InnerText2.default, {
      title: title,
      size: size,
      className: className + "-text",
      data: newData,
      active: active
    })
  );
}

Donut.propTypes = {
  title: _propTypes2.default.string,
  size: _propTypes2.default.number,
  className: _propTypes2.default.string,
  data: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    name: _propTypes2.default.string.isRequired,
    value: _propTypes2.default.number.isRequired
  })).isRequired,
  active: _propTypes2.default.number,
  onHover: _propTypes2.default.func,
  colors: _propTypes2.default.arrayOf(_propTypes2.default.string),
  innerRadius: _propTypes2.default.number,
  outerRadius: _propTypes2.default.number,
  activeOffset: _propTypes2.default.number
};

Donut.defaultProps = {
  title: "Donut Slices",
  size: 500,
  className: "donut-chart",
  colors: ["rgb(51, 102, 204)", "rgb(220, 57, 18)", "rgb(255, 153, 0)", "rgb(16, 150, 24)", "rgb(153, 0, 153)"],
  active: -1,
  // eslint-disable-next-line
  onHover: function onHover(item) {
    return console.log("Hover", item);
  },
  innerRadius: 0.65,
  outerRadius: 0.95,
  activeOffset: 0.05
};
