// react-svg-donut/build/path.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = path;
function coordinates(half, radius, startAngle, endAngle) {
  var startAngleDegrees = Math.PI * startAngle / 180;
  var x1 = half + half * radius * Math.cos(startAngleDegrees);
  var y1 = half + half * radius * Math.sin(startAngleDegrees);
  var endAngleDegrees = Math.PI * endAngle / 180;
  var x2 = half + half * radius * Math.cos(endAngleDegrees);
  var y2 = half + half * radius * Math.sin(endAngleDegrees);

  return {
    x1: x1,
    y1: y1,
    x2: x2,
    y2: y2
  };
}

function arc(width, radius, largeArcFlag, x, y) {
  var z = width / 2 * radius;

  return "A" + z + ", " + z + " 0 " + largeArcFlag + " " + x + ", " + y;
}

function path(activeAngle, startAngle, width, innerRadius, outerRadius) {
  var endAngle = startAngle + activeAngle;

  var largeArcFlagOuter = activeAngle > 180 ? "1 1" : "0 1";
  var largeArcFlagInner = activeAngle > 180 ? "1 0" : "0 0";
  var half = width / 2;
  var outerCoords = coordinates(half, outerRadius, startAngle, endAngle);
  var innerCoords = coordinates(half, innerRadius, startAngle, endAngle);

  var outerArc = arc(width, outerRadius, largeArcFlagOuter, outerCoords.x2, outerCoords.y2);
  var innerArc = arc(width, innerRadius, largeArcFlagInner, innerCoords.x1, innerCoords.y1);

  return "M" + outerCoords.x1 + "," + outerCoords.y1 + "\n  " + outerArc + "\n  L" + innerCoords.x2 + "," + innerCoords.y2 + "\n  " + innerArc + " z";
}
